src/parsing.jl

```diff
@@ -35,7 +35,7 @@ string. `pos` is the 1-based byte offset from which to begin parsing `src`.
 
 See also [`Revise.parse_source`](@ref).
 """
-function parse_source!(mod_exprs_sigs::ModuleExprsSigs, src::AbstractString, filename::AbstractString, mod::Module; mode::Symbol=:sigs)
+function parse_source!(mod_exprs_sigs::ModuleExprsSigs, src::AbstractString, filename::AbstractString, mod::Module; kwargs...)
```

Comment:
Do you expect to call this? My understanding is you're intercepting the method on line 22. If we can change line 27 to
```diff
diff --git a/src/parsing.jl b/src/parsing.jl
index 32ef509..34b9019 100644
--- a/src/parsing.jl
+++ b/src/parsing.jl
@@ -24,7 +24,7 @@ function parse_source!(mod_exprs_sigs::ModuleExprsSigs, filename::AbstractString
         @warn "$filename is not a file, omitting from revision tracking"
         return nothing
     end
-    parse_source!(mod_exprs_sigs, read(filename, String), filename, mod; kwargs...)
+    parse_source!(mod_exprs_sigs, read(filename, String)::String, String(filename)::String, mod; kwargs...)
 end
 
 """
```
(together with the `@nospecialize(ex)` fix later in this file) then that gets rid of the main new source of invalidation risk.

Reply:
Good catch. can you re-run CI?